File: backend/app/application/dto/feeding_dto.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from app.domain.value_objects.feeding_type import FeedingType


@dataclass
class CreateFeedingDTO:
    baby_id: UUID
    feeding_type: FeedingType
    started_at: datetime
    ended_at: datetime | None = None
    amount_ml: int | None = None
    duration_minutes: int | None = None
    memo: str | None = None


@dataclass
class FeedingResponseDTO:
    id: UUID
    baby_id: UUID
    feeding_type: FeedingType
    started_at: datetime
    ended_at: datetime | None
    amount_ml: int | None
    duration_minutes: int | None
    memo: str | None
    created_at: datetime

File: zzippu_ios/zzippu/Data/Network/DTOs/FeedingDTO.swift
// Data/Network/DTOs/FeedingDTO.swift
// 서버 FeedingResponse / FeedingCreateRequest / FeedingUpdateRequest 대응

import Foundation

// MARK: - Response DTO

struct FeedingResponseDTO: Decodable {
    let id: UUID
    let babyId: UUID
    let feedingType: String         // FeedingType raw value
    let startedAt: Date             // datetime — 자동 디코딩
    let endedAt: Date?
    let amountMl: Int?
    let durationMinutes: Int?
    let memo: String?
    let didVomit: Bool?          // 구서버 응답엔 없을 수 있어 옵셔널(매퍼에서 ?? false)
    let createdAt: Date
}

// MARK: - Create Request DTO

struct FeedingCreateRequestDTO: Encodable {
    let feedingType: String
    let startedAt: Date
    let endedAt: Date?
    let amountMl: Int?
    let durationMinutes: Int?
    let memo: String?
    let didVomit: Bool
}

// MARK: - Update Request DTO

struct FeedingUpdateRequestDTO: Encodable {
    let feedingType: String?
    let startedAt: Date?
    let endedAt: Date?
    let amountMl: Int?
    let durationMinutes: Int?
    let memo: String?
    let didVomit: Bool?
}
